### ISSUE TITLE
Compilation errors in generated code when using <daos/> with <pojosAsJavaRecordClasses/>

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12442]
This code generator configuration produces trouble:

```xml
<generate>
    <pojos>true</pojos>
    <pojosAsJavaRecordClasses>true</pojosAsJavaRecordClasses>
    <pojosToString>false</pojosToString>
    <records>true</records>
    <daos>true</daos>
</generate>
```

For the table:

```sql
CREATE TABLE records_3 (i INT, j INT, s VARCHAR(50), CONSTRAINT pk_records_3 PRIMARY KEY (i, j));
```

A `DAO` like this is produced:

```java
public class Records_3Dao extends DAOImpl<Records_3Record, org.jooq.codegen.test.pojosasjavarecords.db.tables.pojos.Records_3, Record2<Integer, Integer>> {

    // [...]

    @Override
    public Record2<Integer, Integer> getId(org.jooq.codegen.test.pojosasjavarecords.db.tables.pojos.Records_3 object) {
        return compositeKeyRecord(object.getI(), object.getJ());
    }

    // [...]
}
```

But there are no such getters on `object`. The correct code would be:

```java
        return compositeKeyRecord(object.i(), object.j());
```

**Comment by lukaseder** [comment]
A fix has been suggested here: https://github.com/jOOQ/jOOQ/pull/12441

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.16.0 and 3.15.3 (https://github.com/jOOQ/jOOQ/issues/12443)

**Comment by lukaseder** [comment]
Neither `KotlinGenerator` nor `ScalaGenerator` are affected by this, given that we do generate ordinary getters (and setters) on `data class` and `case class` POJOs
